src/yokozuna.erl

```diff
@@ -27,6 +27,26 @@
 %%% API
 %%%===================================================================
 
+%% @doc Index a Riak object, given a reason and partition under which
+%%      the object is being indexed.  The object pair contains the new
+%%      and old objects, in the case where a read-before-write resulted
+%%      in an old object we can use to delete siblings before updating.
+-spec index(object_pair(), write_reason(), p()) -> ok.
```

Comment:
I like these doc comments and specs. However, I think they would be even better if they were done as callback definitions in a behavior module, as per my comment on https://github.com/basho/riak_kv/pull/1571